pkg/types/snapstore.go

```diff
@@ -169,3 +180,17 @@ func (c *SnapstoreConfig) Validate() error {
 func (c *SnapstoreConfig) Complete() {
 	c.Prefix = path.Join(c.Prefix, backupFormatVersion)
 }
+
+// CompleteWithOther completes the config based on other config
+func (c *SnapstoreConfig) CompleteWithOther(other *SnapstoreConfig) {
+	if c.Provider == "" {
+		c.Provider = other.Provider
+	}
+	if c.Prefix == "" {
+		c.Prefix = other.Prefix
+	} else {
+		c.Prefix = path.Join(c.Prefix, backupFormatVersion)
+	}
+	c.MaxParallelChunkUploads = other.MaxParallelChunkUploads
+	c.TempDir = other.TempDir
```

Comment:
Now that the source store also has `MaxParallelChunkUploads` and `TempDir` these should also be checked if they exist and only then defaulted to `other.MaxParallelChunkUploads` and `other.TempDir`